### PR TITLE
Feature/moduel 5 hooks

### PR DESCRIPTION
**Pull request by mayurkamaliya**
Implement movie list page
Now it's time to wire our components up and build the real app.
Create a "MovieListPage" component. You will place all the markup and logic for the main page here.
Using useState hook define several states for the component:
search query
sort criterion
active genre
movie list (array)
selected movie (default to undefined or null)

Render components that you've previously built to create the movie list page. It should look similar to the [design prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/%5BCDP%5D-Home-Task-%E2%80%93-React-v1). Pixel-perfect design is not necessary, but overall composition and placement should look the same.
Pass search query state to the SearchForm component. Modify the state when the search form is submitted.
If selected movie is defined, instead of SearchForm render MovieDetails and pass the selected movie in props.
Pass static genre list and an active genre to the GenreSelect component. Modify the active genre state when other genres are selected.
Pass current sort criterion to SortControl. Modify sort criterion state when the user changes sorting.
Render MovieTile for every movie in state. Pass movie info to the MovieTile. You can define a mock movie array for now, so that your array is not empty and you can test results. Handle click events on MovieTile and update the selected movie state when a movie is selected.

As a result of this step, you should have a movie list page, with all controls working but without backend integration.

## Visuals
On genre select fetching movies: 
2023-11-10 12_16_54-Creds - OneNote

2023-11-10 12_18_58-React App

Sorting:
2023-11-10 12_19_45-Creds - OneNote

Searching movie:
2023-11-10 12_20_38-Creds - OneNote

### CODE REVIEW

File: src/components/Movies/MovieList/MovieList.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  };
+
+  const handleNextPage = () => {
+    setOffset(offset + limit);
+  };
+
+  return (
+    <div>
+      <section className="movielist">
+        {moviesResponse !== null &&
```
Comment (andreyBalabukha) [inline src/components/Movies/MovieList/MovieList.jsx:57]: I think we can use here just `movieResponse &&`

File: src/components/Movies/MovieList/MovieList.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                film={input}
+              />
+            </article>
+          ))}
+      </section>
+      <div className="pagination">
+        <button
+          onClick={handlePreviousPage}
+          className="pagination-button"
+          disabled={offset === 0}
```
Comment (andreyBalabukha) [inline src/components/Movies/MovieList/MovieList.jsx:78]: `useMemo` would be more suitable here

File: src/components/Movies/MovieList/MovieList.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        >
+          Previous
+        </button>
+        <div className="page-numbers">
+          Page {currentPage} of {totalPages}
+        </div>
+        <button
+          onClick={handleNextPage}
+          className="pagination-button"
+          disabled={offset + limit >= totalMovies}
```
Comment (andreyBalabukha) [inline src/components/Movies/MovieList/MovieList.jsx:88]: `useMemo` hook would be more suitable here

File: src/components/SortAndGenreControl/SortControl.jsx [hunk 4]
```diff
@@ -10,14 +11,17 @@ class SortControl extends React.Component {
     const { currentSelection } = this.props;
 
     return (
-      <div>
-        <label htmlFor="sortSelect">Sort by : </label>
+      <div className="sort-container">
+        <label className="label" htmlFor="sortSelect">
+          Sort by :{" "}
```
Comment (andreyBalabukha) [inline src/components/SortAndGenreControl/SortControl.jsx:16]: Why do you use here `{" "}`? Could be replaced with the constant, like
`const blankString = " "`

File: src/components/Movies/MovieForm/MovieForm.jsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    this.state = {
+      formData: props.initialMovieInfo || {
+        name: "",
+        year: "",
+        movieUrl: "",
+        vote_average: "",
+        genres: "",
+        runtime: "",
+        overview: "",
+      },
```
Comment (andreyBalabukha) [inline src/components/Movies/MovieForm/MovieForm.jsx:18]: can use here the default props:
https://blog.logrocket.com/complete-guide-react-default-props/

**Review by andreyBalabukha — APPROVED** [review]
Looks good to me with the minor comments

File: src/components/Movies/MovieList/MovieList.jsx (hunk 1 above)
Comment (mayurkamaliya) [reply src/components/Movies/MovieList/MovieList.jsx]: Removed !== null code.

File: src/components/Movies/MovieList/MovieList.jsx (hunk 2 above)
Comment (mayurkamaliya) [reply src/components/Movies/MovieList/MovieList.jsx:82]: Used useMemo in handlePrevousPage.

File: src/components/Movies/MovieList/MovieList.jsx (hunk 3 above)
Comment (mayurkamaliya) [reply src/components/Movies/MovieList/MovieList.jsx:92]: Used useMemo in handleNextPage.

File: src/components/SortAndGenreControl/SortControl.jsx (hunk 4 above)
Comment (mayurkamaliya) [reply src/components/SortAndGenreControl/SortControl.jsx]: Removed `{" "}` and used BlankString constant.

File: src/components/Movies/MovieForm/MovieForm.jsx (hunk 5 above)
Comment (mayurkamaliya) [reply src/components/Movies/MovieForm/MovieForm.jsx]: Removed the defaultProps declaration from this.state and added MovieForm.defaultProps.

**Pull request merged**